### ISSUE TITLE
Help wanted

### ISSUE THREAD

**Issue by hesamkh** [facebookresearch/RandomizedValueFunctions#1]
Hi,

I have some questions about the code(code for "Randomized Value Functions via Multiplicative Normalizing Flows" paper). I want to use this code for my environment and I can run it for my environment without any error but I have some problems:
1.My states` size are [100,10] and I change "nn.Conv2d" in "local_mnf_agent" to "nn.Linear" because my states are 2 dim not 3 dim!. Should I exert any  other change to code? for example in "MNFLinear" function?
2.My action is [100,2] and I remove ".data.max(1)[1][0]" in "agent.act" to get [100,2] action. (the original code just release one action in each iteration.) But after some iteration the actions are reiterative and agent return 38 for first column and 14 for second column(the max possible amount) and I don`t know why it is happen? because this action reduce the reward and the further I go forward and with increasing repetition the reward increases to zero!!

I have used my environment with other RL algorithms and I am sure it dosen`t have any problem. I think I should exert other changes to adopt this code with my environment.
I'll be happy if you help me

**Comment by ahmed-touati** [comment]
Hi @hesamkh,
1) you can use ```MNFLinear``` if you would like to define stochastic layer. Otherwise, you can use ```nn.Linear```
2) I don't understand what do you mean by ``` My action is [100,2] ```. Could you explain?  
If you remove ```.data.max(1)[1][0]```, you would get the Q values of all actions. 

Maybe as sanity check, you could try to adapt and run dqn_agent first on your environment.

**Comment by hesamkh** [comment]
Hi @ahmed-touati 

thank you for your reply!

I mean at each time step agent action is a 100 * 2 array not a single number. In the original code action is a number between 1 to 6.  but in my case action is an 100*2 array and at each time step RL should release 200 number. agent.act in the original code return a number like 4 but I need it returns 100*2 array because I have 100 user and each of them has 2 bandit! In other word my case is a multi agent RL not a single agent as in original code.

**Comment by ahmed-touati** [comment]
Hi @hesamkh,

The current implementation supports only discret actions and single agent like DQN does.
